File: benchmark/template/programs/go/adversarial/ADV-13_single.go
package main

import "fmt"

type A struct{ v int64 }

type B struct{ v int64 }

func main() {
	fmt.Println("ADV-START")
	var r any = A{v: 42}
	b := r.(B)
	fmt.Printf("OBS=V:%d\n", b.v)
	fmt.Println("ADV-END")
}

File: benchmark/template/programs/cpp/adversarial/ADV-13_single.cpp
#include <cstdio>
#include <cstdint>

struct A { std::int64_t v; };
struct B { std::int64_t v; };

int main() {
    printf("ADV-START\n"); fflush(stdout);
    A a{42};
    void* p = &a;
    B* b = static_cast<B*>(p);
    printf("OBS=V:%lld\n", (long long)b->v); fflush(stdout);
    printf("ADV-END\n"); fflush(stdout);
    return 0;
}
